src/components/TodoApp/TodoApp.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import TodoForm from '../TodoForm/TodoForm';
+import Todo from '../Todo/Todo';
+
+export default class TodoApp extends React.Component {
+  state = {
+    todos: [],
+    todosToShow: 'all',
+  }
+
+  addTodo = (todo) => {
+    if (todo.text.trim() !== '') {
+      this.setState(prevState => ({
+        todos: [todo, ...prevState.todos],
+      }));
+    }
+  };
+
+  handleDelete = (id) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.filter(todo => todo.id !== id),
+    }));
+  }
+
+  toggleComplete = (id) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map(todo => (
+        todo.id === id
+          ? {
+            ...todo,
+            complete: !todo.complete,
+          }
+          : { ...todo }
```

Comment:
деструктуризация ничего не делает полезного в этом случае
```suggestion
          : todo
```